urlc175f05cb8_mchllbrt_PermCode_tgz-pJ8-permlib_examples_UniversalSeparableJ8/src/permlib/property/Griddable.java

```diff
@@ -0,0 +1,89 @@
+package permlib.property;
+
+import java.util.Arrays;
+import permlib.utilities.MultisetCodes;
+import permlib.Permutation;
+
+/**
+ * The property of being griddable. That is, a class with an associated matrix
+ * whose entries represent simpler permutation classes. All permutations in the
+ * class can be chopped apart into sections that correspond to the matrix 
+ * entries.
+ * 
+ * @author M Albert
+ */
+public final class Griddable implements PermProperty {
+
+    final PermProperty[][] propertyGrid;
+    int rows;
+    int cols;
+
+    public Griddable(PermProperty[][] propertyGrid) {
+        computeGridSize(propertyGrid); // Allow for lazy users who can't be bothered to pad rows with Empty()
+        this.propertyGrid = new PermProperty[rows][cols];
+        for (int r = 0; r < rows; r++) {
+            for (int c = 0; c < cols; c++) {
+                if (r < propertyGrid.length && c < propertyGrid[r].length) {
+                    this.propertyGrid[r][c] = propertyGrid[r][c];
+                } else {
+                    this.propertyGrid[r][c] = PermProperty.EMPTY;
+                }
+            }
+        }
+    }
+
+    private void computeGridSize(Object[][] g) {
+        this.rows = g.length;
+        this.cols = 0;
+        for (int r = 0; r < rows; r++) {
+            this.cols = Math.max(this.cols, g[r].length);
+        }
+    }
+
+    @Override
+    public final boolean isSatisfiedBy(Permutation p) {
+        return isSatisfiedBy(p, false);
+    }
+    public final boolean isSatisfiedBy(Permutation p, boolean verbose) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.